controllers/dogu_controller.go

```diff
@@ -169,59 +197,77 @@ func (r *doguReconciler) handleSupportMode(ctx context.Context, doguResource *k8
 	return nil, nil
 }
 
-func (r *doguReconciler) evaluateRequiredOperation(ctx context.Context, doguResource *k8sv1.Dogu) (operation, error) {
+func (r *doguReconciler) evaluateRequiredOperations(ctx context.Context, doguResource *k8sv1.Dogu) ([]operation, error) {
 	logger := log.FromContext(ctx)
 	if doguResource.DeletionTimestamp != nil && !doguResource.DeletionTimestamp.IsZero() {
-		return Delete, nil
+		return []operation{Delete}, nil
 	}
 
+	var err error
+	var operations []operation
 	switch doguResource.Status.Status {
 	case k8sv1.DoguStatusNotInstalled:
-		return Install, nil
+		return []operation{Install}, nil
 	case k8sv1.DoguStatusPVCResizing:
-		return ExpandVolume, nil
-	case k8sv1.DoguStatusInstalled:
-		isVolumeExpansion, err := r.checkForVolumeExpansion(ctx, doguResource)
+		operations = append(operations, ExpandVolume)
+		operations, err = r.appendRequiredPostInstallOperations(ctx, doguResource, operations)
 		if err != nil {
-			return Ignore, err
+			return nil, err
 		}
-
-		if isVolumeExpansion {
-			return ExpandVolume, nil
+	case k8sv1.DoguStatusInstalling:
```

Comment:
`k8sv1.DoguStatusUpgrading` needs this behaviour too

Reply:
Thanks for finding this! This would've been one big bad bug otherwise.